Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -1656,6 +1650,41 @@ static Ref<CSSValueList> animationShorthandValue(CSSPropertyID property, const A
     return CSSValueList::createCommaSeparated(WTFMove(parentList));
 }
 
+static Ref<CSSValue> singleTransitionValue(const Animation& transition)
+{
+    static NeverDestroyed<Ref<TimingFunction>> initialTimingFunction(Animation::initialTimingFunction());
+
+    // If we have a transition-delay but no transition-duration set, we must serialze
+    // the transition-duration because they're both <time> values and transition-delay
+    // comes first.
+    auto showsDelay = transition.delay() != Animation::initialDelay();
+    auto showsDuration = showsDelay || transition.duration() != Animation::initialDuration();
+
+    CSSValueListBuilder list;
+    if (transition.property() != Animation::initialProperty())
+        list.append(createTransitionPropertyValue(transition));
+    if (showsDuration)
+        list.append(valueForAnimationDuration(transition.duration()));
+    if (auto* timingFunction = transition.timingFunction(); *timingFunction != initialTimingFunction.get())
+        list.append(valueForAnimationTimingFunction(*timingFunction));
+    if (showsDelay)
+        list.append(valueForAnimationDelay(transition.delay()));
+    if (transition.allowsDiscreteTransitions() != Animation::initialAllowsDiscreteTransitions())
+        list.append(valueForTransitionBehavior(transition.allowsDiscreteTransitions()));
+    return CSSValueList::createSpaceSeparated(WTFMove(list));
+}
+
+static Ref<CSSValue> transitionShorthandValue(const AnimationList* transitions)
+{
+    if (!transitions || transitions->isEmpty())
+        return CSSPrimitiveValue::create(CSSValueAll);
+
+    CSSValueListBuilder list;
+    for (auto& transition : *transitions)
+        list.append(singleTransitionValue(transition));
+    return CSSValueList::createCommaSeparated(WTFMove(list));
```

Comment:
Can we assert that the list isn't empty?

Reply:
The `CSSValueListBuilder` or resulting `CSSValueList`? I guess both will have the same number of items.

Reply:
CSSValueListBuilder

Reply:
but either is fine

Reply:
This review feedback got me to think about `singleTransitionValue()` yielding an empty list for an `all` value and sure enough it did, so that's a bug to fix and additional test to write. Will file a new PR with those changes and the suggested assertion.

Reply:
The `all` case was fixed in a subsequent version of this patch, and tested.